harvesting/esciencecenter.py

```diff
@@ -0,0 +1,469 @@
+        if not "primaryKey" in related_organization or ("logo" in related_organization and related_organization["logo"]["data"] == ""):
+            related_organization = add_new_organization_to_database(token,api_organizations_data,related_organization_name)
+        related_organizations.append({"foreignKey":related_organization["primaryKey"]})
+    return(related_organizations)
+
+
+def get_source_related_projects(project_data_source,api_projects_data):
+    related_projects = []
+    for related_project_title in project_data_source["related_project_data"]:
+        related_project = get_data_by_title(api_projects_data,related_project_title)
+        if not "primaryKey" in related_project:
+            logger.warning(f"unknown project [1] {related_project_title}")
+        else:
+            related_projects.append({"foreignKey":related_project["primaryKey"]})
+    return(related_projects)
+
+
+def update_project_data(token,api_projects_data,api_organizations_data,api_persons_data,project_data_source,project_data_target):
+    project_data_target["description"] = remove_html_links(project_data_source["text"])
+    project_data_target["imageUrl"] = project_data_source["imageUrl"]
+    project_data_target["updatedBy"] = "esciencecenter.nl.crawler"
+    project_data_target["updatedAt"] = get_current_time_string()
+    project_data_target["related"]["organizations"] = get_source_related_organizations(token,project_data_source,api_organizations_data)
+    project_data_target["related"]["projects"] = get_source_related_projects(project_data_source,api_projects_data)
+    project_data_target["team"] = get_source_team(token,project_data_source,api_persons_data,project_data_target["related"]["organizations"])
+    return(project_data_target)
+
+
+def update_project_in_database(token,api_projects_data,api_organizations_data,api_persons_data,project_data_source,project_data_target):
+    project_data_target = update_project_data(token,api_projects_data,api_organizations_data,api_persons_data,project_data_source,project_data_target)
+    project_api_url = "https://localhost/api/project/"+project_data_target["primaryKey"]["id"]
+    project_api_response = requests.put(project_api_url,json=project_data_target,headers={'Authorization': 'Bearer %s' % token},verify=False)
+    if request_was_successful(project_api_response):
+        print(f'updated project "{project_data_source["title"]}"')
+        return(True)
+    else:
+        logger.warning(f'failed updating project in database: {project_api_response} {project_data_source["title"]}')
+        return(False)
+
+
+def find_project_in_database(token,api_projects_data,project_data_source):
+    for project_data_target in api_projects_data:
+        if project_data_source["title"] == project_data_target["title"]:
+            return(project_data_target)
+    return(add_new_project_to_database(token,api_projects_data,project_data_source))
```

Comment:
can you add a `print` with the name of the project that will be added here please

Reply:
also for other entities that are added, I think the script is generating some duplicates, particularly in the person collection.

Reply:
The names of all newly added entities are already printed. The print lines can be found in the code by searching for "added new". For example, names of added projects are printed at line 345.

Reply:
Perhaps I'm not seeing this because the sample data was updated already. My output is the same as Stefan's above, i.e. `added new organization "Software for Chemicals & Materials"` is the only thing newly added.

Reply:
The program only uses the names of persons so when these are not consistently spelled, duplicates will be generated. It is indeed a good idea to check for these. The present database also contains a few.

Reply:
Maybe tomorrow, I'll try to see what the output looks like if I use old sample data

Reply:
the PR is now based on master's sample data. marking as resolved